sidecarhttp/envoy_api.go

```diff
@@ -354,8 +366,26 @@ func (s *EnvoyApi) EnvoyListenerFromService(svc *service.Service, port int64) *E
 					},
 				},
 			},
-		},
+		}
+	} else { // == "tcp"
```

Comment:
Should be explicit check?

Reply:
@relistan I was thinking about this too. We don't do any validation for the label value [here](https://github.com/Nitro/sidecar/blob/8356113484229fcb74b085760d4e3bc7e85a1aa5/service/service.go#L134-L138), so if the user puts some other value in there, they might be surprised to get a TCP proxy.

Does HAProxy support any other proxy modes? Otherwise, I'd be inclined to refactor the code a bit both here and in `service.go` and explicitly limit ProxyMode to either `http` or `tcp`.

Maybe we should take this as part of the refactor work for pulling this code out of `envoy-docker-shim`? WDYT?

Reply:
Merging for now and leaving this for a future refactor.